src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXCEPTION_MESSAGE = "Index out of bound";
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private class Node<T> {
+        private T element;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+            size++;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            Node<T> firstNode = new Node<>(null, value, null);
+            head = firstNode;
+            tail = firstNode;
+        } else {
+            Node<T> nextNode = new Node<>(tail, value, null);
+            tail.next = nextNode;
+            tail = nextNode;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        indexIsValidAdd(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
```

Comment:
```suggestion
            add(value);
            return;
        } 
        if (index == 0) {
```

it's hard to read complicated `if-else` constructions